File: C4_InnerTube/InnerTubeCS/SharedUtilities/iTunesSync.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using iTunesLib;
using System.IO;

namespace SharedUtilities
{
    public class iTunesSync : IVideoService
    {
        iTunesApp iTunes = new iTunesApp();


        #region IVideoService Members

        public void Sync(string filePath)
        {
            //only get MP4 files
            string[] fileList = Directory.GetFiles(filePath, "*.mp4",SearchOption.TopDirectoryOnly);
            try
            {
                foreach (var f in fileList)
                {
                    //Add file                    
                    iTunes.LibraryPlaylist.AddFile(f);

                    

                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("iTunes error: " + ex.Message); 
            }
            finally
            {
                this.iTunes = null; 
            }            
        }

        #endregion

        public void UpdateIPod()
        {
            iTunes.UpdateIPod();
        }


        ~iTunesSync()
        {
            if (this.iTunes != null)
            {
                //cleanup
                this.iTunes = null; 
            }
            

        }
    }
}

File: C04_InnerTube/InnerTubeVB/SharedUtilities/iTunesSync.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.IO
Imports iTunesLib

Namespace SharedUtilities
	Public Class iTunesSync
		Implements IVideoService
		Private iTunes As New iTunesApp()

		#Region "IVideoService Members"

		Public Sub Sync(ByVal filePath As String) Implements IVideoService.Sync
			'only get MP4 files
			Dim fileList() As String = Directory.GetFiles(filePath, "*.mp4",SearchOption.TopDirectoryOnly)
            Try
                For Each f In fileList
                    'Add file
                    iTunes.LibraryPlaylist.AddFile(f)
                Next f
            Catch ex As Exception
                System.Diagnostics.Debug.WriteLine("iTunes error: " & ex.Message)
            Finally
                Me.iTunes = Nothing
            End Try
        End Sub

		#End Region

		Public Sub UpdateIPod()
			iTunes.UpdateIPod()
		End Sub

		Protected Overrides Sub Finalize()
            If Me.iTunes IsNot Nothing Then
                'cleanup 
                Me.iTunes = Nothing
            End If

		End Sub
	End Class
End Namespace
